docs/programming_guide/experiment_tracking.rst

```diff
@@ -0,0 +1,131 @@
+.. _experiment_tracking:
+
+###################
+Experiment Tracking
+###################
+
+***********************
+Overview and Approaches
+***********************
+
+In a federated computing setting, the data is distributed across multiple devices or systems, and training is run
+on each device independently while preserving each client's data privacy.
+
+Assuming a federated system consisting of one server and many clients and the server coordinating the ML training of clients,
+we can interact with ML experiment tracking tools in two different ways: 
+
+    - Client-side experiment tracking: Each client will directly send the log metrics/parameters to the ML experiment tracking server or local file system
+    - Server-side experiment tracking: Clients will send the log metrics/parameters to FL server, and the FL server will send the metrics to ML experiment
+      tracking server or local file system
+
+Each approach will have its use cases and unique challenges. In NVFLARE, we developed a server-side approach:
+
+    - Clients don't need to have access to the tracking server, avoiding the additional
+      authentication for every client. In many cases, the clients may be from different organizations
+      and different from the host organization of the experiment tracking server. 
+    - Since we reduced connections to the tracking server from N clients to just one server, the traffic to the tracking server 
+      can be highly reduced. In some cases, such as in MLFLow, the events can be buffered in the server and sent to the tracking
+      server in batches, further reducing the traffic to the tracking server. The buffer may add additional latency, so you can
+      disable the buffering if you can set the buffer flush time to 0 assuming the tracking server can take the traffic.
+    - Another key benefit of using server-side experiment tracking is that we separate the metrics data collection 
+      from the metrics data delivery to the tracking server. Clients are only responsible for collecting metrics, and only the server needs to 
+      know about the tracking server. This allows us to have different tools for data collection and data delivery.
+      For example, if the client has training code with logging in Tensorboard syntax, without changing the code, the server can
+      receive the logged data and deliver the metrics to MLflow.
+    - Server-side experiment tracking also can organize different clients' results into different experiment runs so they can be easily
+      compared side-by-side. 
+
+**************************************
+Tools, Sender, LogWriter and Receivers
+**************************************
+
+With the "experiment_tracking" examples in the advanced examples directory, you can see how to track and visualize
+experiments in real time and compare results by leveraging several experiment tracking tools:
+
+    - Tensorboard
```

Comment:
It will be good to make each of these a link to their official website so people knows 100% what these are